src/EFCore.PG/Storage/Internal/NpgsqlDatabaseCreator.cs

```diff
@@ -158,12 +158,34 @@ protected override async Task<bool> HasTablesAsync(CancellationToken cancellatio
             => (bool)(await CreateHasTablesCommand().ExecuteScalarAsync(_connection, cancellationToken: cancellationToken));
 
         IRelationalCommand CreateHasTablesCommand()
-            => _rawSqlCommandBuilder
-                .Build(@"
-                    SELECT CASE WHEN COUNT(*) = 0 THEN FALSE ELSE TRUE END
-                    FROM information_schema.tables
-                    WHERE table_type = 'BASE TABLE' AND table_schema NOT IN ('pg_catalog', 'information_schema')
-                ");
+        {
+            var schemaInModel = Dependencies.Model.GetEntityTypes()
+                .Select(entityType => entityType.Relational().Schema)
+                .Where(schema => !string.IsNullOrEmpty(schema))
+                .Distinct()
+                .ToArray();
+
+            return schemaInModel.Length == 0
+                ? CreateHasAnyTablesCommand() // case when no schema (default or per table) has been defined in the model
+                : CreateHasAtLeastOneTableWithSchemaCommand(schemaInModel);
+        }
+
+        IRelationalCommand CreateHasAtLeastOneTableWithSchemaCommand(string[] schemaInModel)
+        {
+            var sql = string.Format(@"
+                SELECT CASE WHEN COUNT(*) = 0 THEN FALSE ELSE TRUE END
+                FROM information_schema.tables
+                WHERE table_type = 'BASE TABLE' AND table_schema IN ('{0}')",
+                string.Join("','", schemaInModel));
```

Comment:
IMHO they are excluded, the clause is positive test `IN`

Reply:
I'm thinking of the case where a system table is included in the model:
```c#
builder.ToTable("tables", "information_schema");
```

The preexistence of a system table would cause `HasTables` to always return `true`, even when other model schemas do not exist.